lib/node_modules/@stdlib/blas/base/wasm/zscal/benchmark/benchmark.module.main.js

```diff
@@ -0,0 +1,141 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var hasWebAssemblySupport = require( '@stdlib/assert/has-wasm-support' );
+var Memory = require( '@stdlib/wasm/memory' );
+var bytesPerElement = require( '@stdlib/ndarray/base/bytes-per-element' );
+var uniform = require( '@stdlib/random/array/uniform' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var pow = require( '@stdlib/math/base/special/pow' );
+var Float64Array = require( '@stdlib/array/float64' );
+var pkg = require( './../package.json' ).name;
+var zscal = require( './../lib' );
+
+
+// VARIABLES //
+
+var opts = {
+	'skip': !hasWebAssemblySupport()
+};
+var options = {
+	'dtype': 'float64'
+};
+
+
+// FUNCTIONS //
+
+/**
+* Creates a benchmark function.
+*
+* @private
+* @param {PositiveInteger} N - array length
+* @returns {Function} benchmark function
+*/
+function createBenchmark( N ) {
+	return benchmark;
+
+	/**
+	* Benchmark function.
+	*
+	* @private
+	* @param {Benchmark} b - benchmark instance
+	*/
+	function benchmark( b ) {
+		var byteOffset;
+		var view;
+		var xptr;
+		var zptr;
+		var mod;
+		var mem;
+		var nb;
+		var i;
+
+		// Create a new BLAS routine interface:
+		mem = new Memory({
+			'initial': 0
+		});
+		mod = new zscal.Module( mem );
+
+		// Initialize the module:
+		mod.initializeSync(); // eslint-disable-line node/no-sync
+
+		// Reallocate the underlying memory to allow storing one vector and a complex number:
+		nb = bytesPerElement( 'complex128' );
+		mod.realloc( ((N*2)*nb)+nb );
+
+		// Define a pointer (i.e., byte offset) to the first vector element:
+		xptr = 0;
+
+		// Define a pointer to a complex number:
+		zptr = (N*2) * nb;
+
+		// Write random values to module memory:
+		mod.write( xptr, uniform( (N*2), -100.0, 100.0, options ) );
+		mod.write( zptr, new Float64Array( [ 1.0, 0.0 ] ) );
+
+		// Retrieve a DataView of module memory:
+		view = mod.view;
+
+		b.tic();
+		for ( i = 0; i < b.iterations; i++ ) {
+			mod.main( (N*2), zptr, xptr, 1 );
+			byteOffset = xptr + ( (i%(N*2))*nb );
```

Comment:
You seem to be a bit confused regarding bytes offsets and the mixing of real and complex.

```suggestion
		mod.realloc( (N*nb)+nb );

		// Define a pointer (i.e., byte offset) to the first vector element:
		xptr = 0;

		// Define a pointer to a complex number:
		zptr = N * nb;

		// Write random values to module memory:
		mod.write( xptr, uniform( (N*2), -100.0, 100.0, options ) );
		mod.write( zptr, new Float64Array( [ 1.0, 0.0 ] ) );

		// Retrieve a DataView of module memory:
		view = mod.view;

		b.tic();
		for ( i = 0; i < b.iterations; i++ ) {
			mod.main( N, zptr, xptr, 1 );
			byteOffset = xptr + ( (i%N)*nb );
```